samples/sample_decode/src/pipeline_decode.cpp

```diff
@@ -1571,6 +1571,13 @@ mfxStatus CDecodingPipeline::SyncOutputSurface(mfxU32 wait)
 
     mfxStatus sts = m_mfxSession.SyncOperation(m_pCurrentOutputSurface->syncp, wait);
 
+    if (MFX_ERR_GPU_HANG == sts) {
+        msdk_printf(MSDK_STRING("GPU hang happened\n"));
+        // Output surface can be corrupted
+        // But should be delivered to output anyway
+        sts = MFX_ERR_NONE;
```

Comment:
Will/how sample_decode stop working after this error? Printing message is cool, but error handling looks strange to me.

Reply:
This was the requirements from library team. By default library and samples will use "soft recovery" - so nothing should be done in case of GPU hang (library will recover by itself), just print error message notifying that GPU hang occurred and several frames might be missing.